doc/tutorial/network_ring.rst

```diff
@@ -114,8 +114,12 @@ The execution
 
 To create a simulation, we must create an :class:`arbor.context` and :py:class:`arbor.domain_decomposition`.
 
-Step **(12)** creates a default execution context, and uses the :func:`arbor.partition_load_balance` to create a
-default domain decomposition. You can print the objects to see what defaults they produce on your system.
+Step **(12)** initalizes the ``threads`` parameter of :class:`arbor.context` with the ``avail_threads`` flag. By supplying
+this flag, a context is constructed that will use all locally available threads. On your local machine this will match the
+number of cpu-cores, times two if your cpu provides some form of simultaneous multithreading. Especially with large numbers
+of cells you will notice the speed-up. :func:`arbor.partition_load_balance` creates a default domain decomposition, which 
```

Comment:
This only is true if you request more cell groups, right? Both the 'distribute' and -- consequently -- the 'speed-up' parts.
See simulation.cpp:357, tasks push updates to whole _groups_.
```
> python3 brunel.py
<arbor.context: num_threads 2, has_gpu False, has_mpi False, num_ranks 1>
<arbor.domain_decomposition: domain_id 0, num_domains 1, num_local_cells 500, num_global_cells 500, groups 500>
```
You'll need to add something like this
```
    hint = arbor.partition_hint()
    hint.cpu_group_size = opt.group_size
    hints = {arbor.cell_kind.lif: hint}
    decomp = arbor.partition_load_balance(recipe, context, hints)
```
to get more groups on one MPI task _or_ partition explicitly.

Reply:
By 'requesting cell groups', do you mean 'requesting ranks'?

What I mean to explain without introducing more jargon than required is that cells, also if they are the same kind/description, will be spread over the available execution threads (when `avail_threads` is set), and consequently a speed-up (compared to a default context) will be experienced.

Reply:
No, you can in fact request more groups using hints (like in the example I gave) or an explicit partitioning similar
to a list of lists of cell `gid`s. However, I realised that's not needed here, the default is one group per cell (here 500, see example) _and not_ one group per task. So, this is ok, but for optimisation one group per thread might be better (something better left out here, but keep in mind!)